services/src/modules/apollo-server-plugins/introspection-query-policy.ts

```diff
@@ -0,0 +1,47 @@
+import { GraphQLResolveInfo, SchemaMetaFieldDef } from 'graphql';
+import { ForbiddenError } from 'apollo-server-core';
+import logger from '../logger';
+import { RequestContext } from '../context';
+import { ResourceGroup } from '../resource-repository';
+import { ActiveDirectoryAuth } from '../upstreams/authentication';
+import { PolicyExecutor } from '../directives/policy';
+
+const INTROSPECTION_QUERY_FIELD_NAMES = new Set(['__schema', '__type']);
+
+function isIntrospectionRequest(info: GraphQLResolveInfo) {
+  return INTROSPECTION_QUERY_FIELD_NAMES.has(info.fieldName);
+}
+
+export function createIntrospectionQueryPolicyPlugin(): void {
+  const originalSchemaMetaFieldResolver = SchemaMetaFieldDef.resolve;
```

Comment:
should be done in global scope to prevent multiple wrapping with policy validation (in tests)
maybe something like this:

```ts
const originalSchemaMetaFieldDefResolve = SchemaMetaFieldDef.resolve;
const originalTypeMetaFieldDefResolve = TypeMetaFieldDef.resolve;

export function createIntrospectionQueryPlugin(): ApolloServerPlugin {
  return {
    serverWillStart() {
      wrapWithPolicyValidation(SchemaMetaFieldDef, originalSchemaMetaFieldDefResolve);
      wrapWithPolicyValidation(TypeMetaFieldDef, originalTypeMetaFieldDefResolve);
    },
  };
}

function wrapWithPolicyValidation(
  field: GraphQLField<unknown, RequestContext>,
  originalResolve?: GraphQLFieldResolver<unknown, RequestContext, Record<string, unknown>>
) {
...
}
```